File: benchmarks/Java/src/richards/WorkerTaskDataRecord.java
/*
 * This benchmark is derived from Mario Wolczko's Java and Smalltalk version of
 * Richards.
 * 
 * It is modified based on the SOM version and to use Java 8 features.
 * License details:
 *   http://web.archive.org/web/20050825101121/http://www.sunlabs.com/people/mario/java_benchmarking/index.html
 */
package richards;


final class WorkerTaskDataRecord extends RBObject {
  private int destination;
  private int count;

  WorkerTaskDataRecord() {
    destination = HANDLER_A;
    count = 0;
  }

  public int getCount() { return count; }
  public void setCount(final int aCount) { count = aCount; }

  public int getDestination() { return destination; }
  public void setDestination(final int aHandler) { destination = aHandler; }
}

File: benchmarks/C++/src/richards/WorkerTaskDataRecord.cpp
#include "WorkerTaskDataRecord.h"

namespace richards {
    WorkerTaskDataRecord::WorkerTaskDataRecord() {
        _destination = HANDLER_A;
        _count = 0;
    }
        
    int WorkerTaskDataRecord::getCount() const {
        return _count; 
    }

    void WorkerTaskDataRecord::setCount(int aCount) { 
        _count = aCount; 
    }

    int WorkerTaskDataRecord::getDestination() const {
        return _destination; 
    }

    void WorkerTaskDataRecord::setDestination(int aHandler) { 
        _destination = aHandler; 
    }
}